tabbycat/results/views.py

```diff
@@ -265,14 +267,28 @@ def add_success_message(self):
         # Default implementation does nothing.
         pass
 
+    def send_email_receipts(self):
+        # For proper error handling for admin/assistants, overwrite this
+        try:
+            send_ballot_receipt_emails_to_adjudicators(DebateResult(self.ballotsub).as_dicts(), self.debate)
+        except (SMTPException, ConnectionError):
+            return False
+        else:
+            return True
+
     def form_valid(self, form):
         self.ballotsub = form.save()
         if self.ballotsub.confirmed:
             self.ballotsub.confirmer = self.request.user
             self.ballotsub.confirm_timestamp = datetime.datetime.now()
             self.ballotsub.save()
+
+            if self.tournament.pref('enable_ballot_receipts'):
+                self.email_receipts = self.send_email_receipts()
```

Comment:
This line doesn't always execute, which can cause the `self.add_success_message()` call to crash, because no other line sets `self.email_receipts`—so if the ballot isn't being confirmed, or ballot receipts are disabled, then `add_success_message()` crashes with an `AttributeError`.

I think this is the right idea, but either we need to guarantee that `self.email_receipts` always exists, or `add_success_message()` should check for its existence using `getattr(self, 'email_receipts', False)`. I think I'd prefer the latter, because it's more future-proof—the only way to _guarantee_ an instance attribute exists is to put it in the constructor, and even then that makes assumptions coupling this method to the corresponding constructor.

Also, can we call it `self.email_receipts_sent` rather than the ambiguous `self.email_receipts`? Thanks!

Reply:
I can still comment here!

Reply:
Oh right finding this was easier than I thought lol, thanks!